lib/player.js

```diff
@@ -3252,6 +3252,83 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     }
   }
 
+  /**
+   * Return a list of image tracks that can be switched to.
+   *
+   * If the player has not loaded content, this will return an empty list.
+   *
+   * @return {!Array.<shaka.extern.Track>}
+   * @export
+   */
+  getImageTracks() {
+    if (this.manifest_) {
+      const imageStreams = this.manifest_.imageStreams;
+      const StreamUtils = shaka.util.StreamUtils;
+      return imageStreams.map((image) => StreamUtils.imageStreamToTrack(image));
+    } else {
+      return [];
+    }
+  }
+
+  /**
+   * Return a Thumbnail object from a image track Id and time.
+   *
+   * If the player has not loaded content, this will return a null.
+   *
+   * @param {number} trackId
+   * @param {number} time
+   * @return {!Promise.<?shaka.extern.Thumbnail>}
+   * @export
+   */
+  async getThumbnails(trackId, time) {
+    if (this.manifest_) {
+      const imageStream = this.manifest_.imageStreams.find(
+          (stream) => stream.id == trackId);
+      if (!imageStream) {
+        return null;
+      }
+      if (!imageStream.segmentIndex) {
+        await imageStream.createSegmentIndex();
+      }
+      const referencePosition = imageStream.segmentIndex.find(time);
+      if (referencePosition == null) {
+        return null;
+      }
+      const reference = imageStream.segmentIndex.get(referencePosition);
+      // This expression is used to detect one or more numbers (0-9) followed
+      // by an x and after one or more numbers (0-9)
+      const match = /(\d+)x(\d+)/.exec(imageStream.tilesLayout);
+      if (!match) {
+        shaka.log.warning('Tiles layout does not contain a valid format ' +
+            ' (columns x rows)');
+        return null;
+      }
+      const fullImageWidth = imageStream.width || 0;
+      const fullImageHeight = imageStream.height || 0;
+      const columns = parseInt(match[1], 10);
+      const rows = parseInt(match[2], 10);
+      const width = fullImageWidth / columns;
+      const height = fullImageHeight / rows;
+      let positionX = 0;
+      let positionY = 0;
+      const totalImages = columns * rows;
```

Comment:
Please add comments throughout this part of code (L#3312-L#3320) to make it easier to understand what's going on there.

Reply:
I just added the comment, is it enough for you?

Reply:
This is great, I'll ask for a couple last things - 
1) can you break the position defining code into steps with accessible var names for readability (like,
const thumbnailTime = time - reference.startTime etc) 
2) Maybe, use an example to walk through the steps? If our time is 5 seconds and our matrix is [some 2x2 matrix example] than this is how position, X, and Y are determined.

Reply:
Done!